compiler/src/dotty/tools/dotc/transform/TailRec.scala

```diff
@@ -174,6 +175,25 @@ class TailRec extends MiniPhase {
             ).transform(rhsSemiTransformed)
         }
 
+        // Is a simple this a simple recursive tail call, possibly with swapped or modified pure arguments
```

Comment:
It would be nice to have a definition in the human language of what is considered an infinite recursion for our purposes.

Reply:
Done